From 397964facb67eab3da7518f4f26785da31acd2ff Mon Sep 17 00:00:00 2001
From: Romain Rigaux <romain@cloudera.com>
Date: Tue, 9 Dec 2014 11:31:53 -0800
Subject: [PATCH 0134/1173] HUE-2152 [pig] Credentials support for HCat in
 editor

Automatically add a 'hcat' credential when using HCat and a secure
cluster.
(cherry picked from commit dbe2b3abc91a633461cd47000c7d1882b5eddfe8)
---
 apps/pig/src/pig/api.py |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/apps/pig/src/pig/api.py b/apps/pig/src/pig/api.py
index 4f0d93d..9f4b903 100644
--- a/apps/pig/src/pig/api.py
+++ b/apps/pig/src/pig/api.py
@@ -106,9 +106,13 @@ class OozieApi:
         params=json.dumps(pig_params),
         files=json.dumps(files),
         archives=json.dumps(archives),
-        job_properties=json.dumps(job_properties),
+        job_properties=json.dumps(job_properties)
     )
 
+    if pig_script.use_hcatalog and get_oozie(self.user).security_enabled:
+      action.credentials = [{'name': 'hcat', 'value': True}]
+      action.save()
+
     action.add_node(workflow.end)
 
     start_link = workflow.start.get_link()
-- 
1.7.9.5

